### PR TITLE
NPC with no matches/lighter fails to use molotov

### PR DESCRIPTION
**Pull request by renlok**
Attempt to fix NPC with no matches/lighter fails to use molotov, loses all turns #14693

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Something tells me this won't work:
* Grenades start with 0 charges, meaning this would disable grenade throwing
* Molotovs don't use up their charge on failure, meaning that they would still fail to use the molotov

A better way to fix it would be:

Check if the item has a `transform` actor, like this:
```
const auto use_fun = used_tool->get_use( "iuse_transform" );
const iuse_transform *actor = dynamic_cast<const iuse_transform *>( use_fun->get_actor_ptr() );
// If actor == nullptr, it doesn't have it, otherwise it does
```

Then, if the item doesn't have the actor, it doesn't require fire (for example, a spear, a knife or a grenade) and thus can be used. If it does have the actor, it has to be checked for `need_fire`. If `need_fire` is 0 (or less), it doesn't need fire and thus can be used. Otherwise it needs a fire amount check:
```
if( actor == nullptr ) {
    return true;
}
if( actor->need_fire <= 0 ) {
    return true;
}
return has_charges( actor->need_fire );
```

**Comment by renlok** [comment]
Ahh OK that makes sense I couldn't figure out where stuff like that was set

File: src/npcmove.cpp [hunk 1]
```diff
@@ -1075,8 +1076,21 @@ bool npc::alt_attack_available()
...
+                has_amount( elem, 1 ) ) {
+            item *used = nullptr;
+            invslice slice = inv.slice();
+            for (size_t i = 0; i < inv.size(); i++) {
+                if (slice[i]->front().type->id == elem) {
+                    used = &(slice[i]->front());
+                }
+            }
+            const auto use_fun = used->get_use( "iuse_transform" );
+            const iuse_transform *actor = dynamic_cast<const iuse_transform *>( use_fun->get_actor_ptr() );
```
Comment (Coolthulhu) [inline src/npcmove.cpp]: This line causes a crash when `use_fun` is null, there needs to be a check for null `use_fun` before this line.

**Comment by Coolthulhu** [comment]
Not updated for a long white, not ready for merging

**Pull request closed without merging**